### PR TITLE
RFC: dcadec: squash me to 7907932d7294511d9ac846e7b034f831504487e9

### PR DESCRIPTION
**Pull request by fritsch**
Open for discussion.

### CODE REVIEW

**Comment by MilhouseVH** [comment]
Shouldn't `videoplayer.supportdtshd` become `audiooutput.supportdtshd` (or even `audiooutput.enabledtshd`, given the new wording)?

**Comment by fritsch** [comment]
We already have a setting dtshdpassthrough in the passthrough settings, so it should not go under the passthrough settings at all. It's something the decoder does for us, thus videoplayer is fine in my eyes.

**Comment by fritsch** [comment]
So, it basically has nothing to do with audiooutput. As all PCM  output settings are handled in the "upper" part where you choose channels. But I might get convinced here.

Let's ask the master of audio @FernetMenta ping

**Comment by MilhouseVH** [comment]
It just seems wrong to have an audio setting in the video section, particularly when logically it would only be enabled when not using passthrough which is configured in the audio setting. It might be technically correct to have this setting in the Video section, but from a user perspective it doesn't make sense.

The name of the setting isn't (IMHO) that important, videoplayer or audiooutput, but where the user can reasonably expect to find this setting *is* important, and I think that should be Audio not Video. Just my opinion of course. :)

**Comment by fritsch** [comment]
Mmh - mmh.

What about enabling it by default when DTS-HD is disabled? And don't show any settings at all? (besides on the PI1, that is probably too slow doing that?)

**Comment by MilhouseVH** [comment]
>  (besides on the PI1, that is probably too slow doing that?)

Yeah, that's the point of the switch as the new library adds a few percent more load in which case the Pi1 is often better off just extracting DTS core from the DTS HD soundtrack.

Although maybe it makes sense to do what you suggest, perhaps with an as.xml override or is this complicating matters (and adding yet another advanced setting?)

**Comment by fritsch** [comment]
Here - I force pushed. I will present it on all platforms, have set it to false by default and made it conflict with dts-hd setting. I renamed it to: audiooutput.supportdtshdcpudecoding

**Comment by fritsch** [comment]
And better - I also fallback to the default when dcadec is not available at all - so we can push that safely when dcadec is enabled in ffmpeg!

File: addons/resource.language.en_gb/resources/strings.po [hunk 1]
```diff
@@ -16669,12 +16669,12 @@ msgctxt "#38103"
...
 msgctxt "#38120"
-msgid "Support 8 channel DTS HD audio"
+msgid "Enable 8 channel DTS HD audio (CPU) decoding"
 msgstr ""
 
-#: system/settings/rbp.xml
+#: system/settings/settings.xml
 msgctxt "#38121"
-msgid "This option supports 8 channel DTS HD but may use more CPU "
+msgid "This option supports 8 channel DTS HD decoding on the CPU. This option is only available when DTS-HD audio decoding is disabled"
```
Comment (MilhouseVH) [inline addons/resource.language.en_gb/resources/strings.po]: Shouldn't this be: "This option is only available when DTS-HD audio **passthrough** is disabled"?

Comment (fritsch) [inline addons/resource.language.en_gb/resources/strings.po]: Jep of course. Need to make another dumpfix ... will push shortly.

**Comment by fritsch** [comment]
Oki - tested and working :-)

**Comment by FernetMenta** [comment]
I don't think we should have a platform independent setting for this

**Comment by fritsch** [comment]
After dcadec is added to ffmpeg - we will need it. Though there will still be platforms, that don't have enough juice. That was my reasoning.

**Comment by FernetMenta** [comment]
> there will still be platforms, that don't have enough juice

that's a specialization and should be considered on those systems only. why would someone with capable system want to disable it?

**Comment by fritsch** [comment]
Cause he only has 2 channel stereo or uses headphones?

2015-05-08 12:25 GMT+02:00 Rainer Hochecker <notifications@github.com>:

> there will still be platforms, that don't have enough juice
>
> that's a specialization and should be considered on those systems only.
> why would someone with capable system want to disable it?
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/popcornmix/xbmc/pull/10#issuecomment-100185396>.
>



-- 
                   Key-ID:     0x1A995A9B
                   keyserver: pgp.mit.edu
==============================================================
Fingerprint: 4606 DA19 EC2E 9A0B 0157  C81B DA07 CF63 1A99 5A9B

**Comment by FernetMenta** [comment]
thanks

**Comment by popcornmix** [comment]
Fine by me. Thanks.

**Pull request merged**